tephra-core/src/main/java/org/apache/tephra/persist/AbstractTransactionLog.java

```diff
@@ -165,26 +203,36 @@ private void sync() throws IOException {
       // prevent writer being dereferenced
       tmpWriter = writer;
 
-      List<Entry> currentPending = getPendingWrites();
-      if (!currentPending.isEmpty()) {
-        tmpWriter.commitMarker(currentPending.size());
-      }
-
-      // write out all accumulated entries to log.
-      for (Entry e : currentPending) {
-        tmpWriter.append(e);
-        entryCount++;
-        latestSeq = Math.max(latestSeq, e.getKey().get());
+      Entry[] currentPending = getPendingWrites();
```

Comment:
Shall we move this statement out of the synchronize block and call it before the sync block and hold on to the `logSequence` object lock instead?

Reply:
hum... nevermind, seems like doesn't gain much.